Automattic/jetpack
# Search should not require user connection

#### Changes proposed in this Pull Request:
A site level connection is enough for Search to work. And actually in previous versions, we only checked site connection, and I think we should keep doing so.

If user want to connect an account, then he could go to My Jetpack page / or if the user doesn't have a subscription, he could get user connected by purchasing search product on the upsell page.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Prepare a disconnected site with Jetpack Search subscription
* Open `/wp-admin/admin.php?page=jetpack-search`
* Click `login` or `Get Jetpack Search` button, but don't authorize on the next page
* Open `/wp-admin/admin.php?page=jetpack-search` again
* Ensure Search dashboard works
* Ensure Search front end works
* Open `/wp-admin/admin.php?page=my-jetpack`
* Ensure Search doesn't require user connection on My Jetpack page

<img width="1161" alt="Screen Shot 2022-07-25 at 4 47 21 PM" src="https://user-images.githubusercontent.com/1425433/180700657-a2cb073a-ad47-4c4b-8606-c4cad185f238.png">

<img width="1139" alt="Screen Shot 2022-07-25 at 4 51 54 PM" src="https://user-images.githubusercontent.com/1425433/180701081-2560d8f5-8257-4691-b281-966911456f8c.png">
